src/GitVersion.Core/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -10,6 +10,6 @@ public static IServiceCollection AddModule(this IServiceCollection serviceCollec
         return serviceCollection;
     }
 
-    public static TService GetServiceForType<TService, TType>(this IServiceProvider serviceProvider) =>
+    public static TService? GetServiceForType<TService, TType>(this IServiceProvider serviceProvider) =>
         serviceProvider.GetServices<TService>().SingleOrDefault(t => t?.GetType() == typeof(TType));
```

Comment:
May I ask you to change the implementation from SingleOrDefault to just Single and returning always a non-null type? it makes it easier to use IMO.

Reply:
I agree that we should try to return a non-nullable here, but I would prefer doing the `null` check ourselves as that makes it possible for us to throw a more meaningful exception than `Single()` would do if no `TType` is found.